### Suicides rates

In [98]:
import pandas as pd

PATH_TO_FILE = './interactive_data.csv'

data = pd.read_csv(PATH_TO_FILE)

data.head()

Unnamed: 0         Intent         Gender            Age  \
0           1  None selected  None selected  None selected   
1           2  None selected  None selected  None selected   
2           3  None selected  None selected  None selected   
3           4  None selected  None selected  None selected   
4           5  None selected  None selected  None selected   

                     Race  Deaths  Population  Rate  
0           None selected   33599   316299978  10.6  
1                   White   22079   197369634  11.2  
2                   Black    7765    38896382  20.0  
3                Hispanic    3007    54049078   5.6  
4  Asian/Pacific Islander     442    16315561   2.7

In [99]:
len(data)

540

In [100]:
#Important attributes for selection
attrs = data.columns[1:5]
print(attrs)

Index(['Intent', 'Gender', 'Age', 'Race'], dtype='object')


In [121]:
def setDefaultAttributes(data, values):
    for attr in data.columns[1:5]:
        values[attr]='None selected'

In [122]:
attributes = {}
setDefaultAttributes(data, attributes)
    
print(attributes)

{'Intent': 'None selected', 'Gender': 'None selected', 'Age': 'None selected', 'Race': 'None selected'}


In [102]:
def findNumberOfDeathsWithAttributes(data, attributes):
    return data.loc[(data[list(attributes)] == pd.Series(attributes)).all(axis=1)]['Deaths'].iloc[0]

In [105]:
def calculatePercentage(number1, number2):
    return number1/number2*100

In [103]:
setDefaultAttributes(data, attributes)

all_deaths = findNumberOfDeathsWithAttributes(data, attributes)

print(all_deaths)

33599


In [104]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Suicide'

suicide_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(suicide_deaths)

21058


In [120]:
print(calculatePercentage(suicide_deaths, all_deaths))

62.67448435965356


### Conclusion
Accorting to the database, we see that 62.67 percent of all deaths have suicide as their intent.

In [107]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Suicide'
attributes['Gender']='Male'

male_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(male_deaths)

18162


In [119]:
print(calculatePercentage(male_deaths, suicide_deaths))

86.24750688574414


In [109]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Homicide'

homicide_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_deaths)

11726


In [118]:
print(calculatePercentage(homicide_deaths, all_deaths))

34.899848209768145


In [111]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Homicide'
attributes['Gender'] = 'Male'
attributes['Age'] = '15 - 34'

homicide_male_mid_age_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_male_mid_age_deaths)

6520


In [112]:
setDefaultAttributes(data, attributes)
attributes['Intent'] = 'Homicide'
attributes['Gender'] = 'Male'
attributes['Age'] = '15 - 34'
attributes['Race'] = 'Black'

homicide_male_mid_age_black_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_male_mid_age_black_deaths)

4312


In [117]:
print(calculatePercentage(homicide_male_mid_age_black_deaths, homicide_male_mid_age_deaths))

66.13496932515336


In [114]:
setDefaultAttributes(data, attributes)
attributes['Intent'] = 'Homicide'
attributes['Gender'] = 'Female'

homicide_female_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_female_deaths)

1791


In [116]:
print(calculatePercentage(homicide_female_deaths, homicide_deaths))

15.273750639604296
